app/stagemanager.js

```diff
@@ -86,26 +86,35 @@ class StageManager extends MyTrello {
     }
 
     closeUnusedStages(data) {
-      var deferred = Q.defer();
-        var stages = data[0].map(value => value['name']),
-            self = this;
+      var deferred = Q.defer(),
+          stages = data[0].map(value => value['name']),
+          closing = [];
 
         data[1].forEach(function(trelloList) {
+          var listDefer = Q.defer();
+          closing.push(listDefer.promise);
             if (!(stages.includes(trelloList.name))) {
                 self.getListCards(trelloList.id)
                 .then(function(d) {
-                    self.closeList(d, trelloList.id).then(deferred.resolve);
-                })
-                .catch(deferred.reject);
-
+                    self.closeList(d, trelloList.id).then(listDefer.resolve);
+                }).fail(function(err){console.log(err.stack)})
+                .catch(listDefer.reject);
             };
         });
-      return deferred.promise;
+
+        Q.all(closing).then(function() {
```

Comment:
I think you could replace from here down with just:

```javascript
return Q.all(all);
```

Since `Q.all()` returns a promise, this should be functionally equivalent to creating and resolving/rejecting your own promise.